File: tools/get_gpu_info.py
"""GPU adapter information tool."""

from tools import tool
from tools._env_helpers import fmt


@tool(
    name="get_gpu_info",
    description="Return GPU adapter information: detected graphics cards.",
    parameters={"type": "object", "properties": {}, "required": []},
)
def get_gpu_info() -> str:
    return fmt("gpu")

File: tools/get_gpu_info.ts
/** GPU adapter information tool. */

import { registerInfoTool } from "./lib/env_helpers.js";

registerInfoTool({
  name: "get_gpu_info",
  description: "Return GPU adapter information: name, VRAM, driver, refresh rate. For NVIDIA GPUs, also includes real-time VRAM usage, GPU/memory utilization, temperature, power draw, and CUDA version via nvidia-smi.",
  category: "gpu",
});
